src/Tools/ExternalAccess/Razor/RazorTestAnalyzerLoader.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+[Export(typeof(RazorTestAnalyzerLoader)), Shared]
+internal class RazorTestAnalyzerLoader
+{
+    private readonly IDiagnosticAnalyzerService _analyzerService;
+    private readonly DiagnosticService _diagnosticService;
+    private readonly IGlobalOptionService _globalOptionService;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public RazorTestAnalyzerLoader(IDiagnosticAnalyzerService analyzerService, IDiagnosticService diagnosticService, IGlobalOptionService globalOptionService)
+    {
+        _analyzerService = analyzerService;
+        _diagnosticService = (DiagnosticService)diagnosticService;
+        _globalOptionService = globalOptionService;
+    }
+
+    public void InitializeDiagnosticsServices(Workspace workspace)
+    {
+        _globalOptionService.SetGlobalOption(InternalDiagnosticsOptions.NormalDiagnosticMode, DiagnosticMode.LspPull);
+        _ = ((IIncrementalAnalyzerProvider)_analyzerService).CreateIncrementalAnalyzer(workspace);
+        _diagnosticService.Register((IDiagnosticUpdateSource)_analyzerService);
```

Comment:
At some point we'll need to decide if this is the appropriate way to do this, or if this should just happen automatically via a default workspace service or something.  However I'm not super familiar with this area - thoughts @jasonmalinowski or @mavasani ?

Reply:
This is for tests, so we're calling `TestWorkspace.Create`. I would love for that to help us, but I don't think Razor's use should influence real product code. I admit though, I was surprised VS Code needed something special (though greatful 😁)

Reply:
Is the TestWorkspace.Create our TestWorkspace? Because we could consider making this something that's done if an optional flag is passed or something like that rather than needing this.

Reply:
(this doesn't seem bad though, as long as it never gets used outside of tests. I appreciate "Test" being put into the class name so it's explicit)

Reply:
That's exactly why I did it :)